[DR-3399] Fix NPE when DRS PostObject endpoint called without expand

okotsopoulos opened the pull request:
https://broadworkbench.atlassian.net/browse/DR-3399

OpenAPI generates a `DRSPassportRequestModel` for us with a **B**oolean (capital b :-) field called 'expand'.

If our DRS [PostObject](https://jade.datarepo-dev.broadinstitute.org/swagger-ui.html#/DataRepositoryService/PostObject) endpoint is called without 'expand' set in this response body (as is the case with calls from DrsHub), we'd get an NPE when trying to convert the Boolean to a boolean.

This PR adjusts the OpenAPI spec for `DRSPassportRequestModel` to default `expand` to false, and added a unit test to reflect this. (Before the fix, the unit test rightfully failed.)

I wasn't sure initially if we wanted our spec to match the [GA4GH spec exactly](https://github.com/ga4gh/data-repository-service-schemas/blob/master/openapi/components/requestBodies/PostObjectBody.yaml), but after discussing with the team @snf2ye reminded me that we do something similar with our DRS [GetObject](https://jade.datarepo-dev.broadinstitute.org/swagger-ui.html#/DataRepositoryService/GetObject) endpoint (supply a sensible default of false for `expand` when unspecified).

okotsopoulos committed:
[DR-3399] Fix NPE when DRS PostObject endpoint called without expand

OpenAPI generates a DRSPassportRequestModel for us with a Boolean (capital b :-) field called 'expand'.
If our DRS PostObject endpoint is called without 'expand' set in this response body (as is the case with calls from DrsHub), we'd get an NPE when trying to convert the Boolean to a boolean.
Added null handling to default to false for consistency with our DRS GetObject endpoint, and updated unit test set-up to reflect this.
(Before the fix, the unit test set-up change rightfully showed NPEs and failing tests.)

okotsopoulos committed:
Merge branch 'develop' into okotsopo/DR-3399-drs-postobject-npe

rushtong reviewed (approved):
Looks reasonable to me 👍🏽

rjohanek reviewed (approved):
looks good!

snf2ye commented:
> Another more straightforward way to go about this would be to adjust the OpenAPI spec to give a default value of false for expand, but I wasn't sure if we wanted our spec to match the GA4GH spec exactly

Ah, yeah, this is tricky. If we're effectively setting a default in the code, then it might be better to make this explicit in the spec.  In fact, it appears we've already strayed from the DRS spec by making "expand" default to false for the getObject endpoint ([DRS Spec](https://github.com/ga4gh/data-repository-service-schemas/blob/master/openapi/components/parameters/Expand.yaml) vs. [TDR definition](https://github.com/DataBiosphere/jade-data-repo/blob/4fdf9f5192fd72d27f45fe01281be1951fe92127/src/main/resources/api/data-repository-openapi.yaml#L3904)). I may be wrong, but this sort of deviation from the spec doesn't worry me -- by providing a default, it makes the code more resilient, but still adheres to the format that the user would expect across data repos. 

Alternatively, if we leave the spec/code as is, we're effectively requiring the boolean to be set (Which does more or less match the spec).

pshapiro4broad commented on src/main/java/bio/terra/service/filedata/DrsService.java:
```diff
@@ -426,14 +426,14 @@ void verifyPassportAuth(UUID snapshotId, DRSPassportRequestModel drsPassportRequ
       String billingSnapshot) {
     SnapshotProject snapshotProject = getSnapshotProject(snapshot.id);
-    int depth = (expand ? -1 : 1);
+    int depth = (Optional.ofNullable(expand).orElse(false) ? -1 : 1);
```
This is OK but I think it's better to perform defaulting at the outermost possible level, and make this parameter `boolean` everywhere else. It looks like there are two outer sources of this value, `DataRepositoryServiceApiController.getObject()` and `DrsService.lookupObjectByDrsIdPassport()`. It looks like the `getObject()` parameter is defaulted via an annotation so that shouldn't be a source of `null`, so the only case that needs a check is in `lookUpObjectByDrsIdPassport()`.

okotsopoulos replied:
As discussed, I changed approach to define the default in our OpenAPI spec.

pshapiro4broad reviewed (approved):
The change looks OK but I think we should be validating user input at a higher level, such as the controller or the outermost service call.

okotsopoulos committed:
Change in approach: DRSPassportRequestModel defaults expand to false

This follows precedence set with our GetObject endpoint: here we also supplemented the published GA4GH DRS spec to define a top-level sensible default.
Added a unit test.

okotsopoulos committed:
Merge branch 'develop' into okotsopo/DR-3399-drs-postobject-npe
